### ISSUE TITLE
x/build: netbsd-amd64-8_0 builder timing out frequently

### ISSUE THREAD

**Issue by bcmills** [golang/go#31858]
Four of the last five failures on the `netbsd-amd64-8_0` builder are timeouts in various tests.
I can't tell whether the builder is slow or the test is triggering some sort of deadlock.

https://build.golang.org/log/a64cd49d7d367f90e6b79ab84714b4c065763168
https://build.golang.org/log/f7f50b085698aa5480d2c34a3e6c250595317bf6
https://build.golang.org/log/7b749c8616a9b7e9a77c7735038673f9a9dc5d37
https://build.golang.org/log/26dd20510233ec659adf0d5987690164d8abaa6a

CC @bsiegert @bradfitz

**Comment by bradfitz** [comment]
@bsiegert, what makes you think that? I suspect the NetBSD port (or NetBSD) is buggy based on looking at historical data over the past month on this builder. The `cgo_test` has a 43.2 second median over 309 runes. And the `cmd/compile/internal/types` test is fast across all builders: 3.4 seconds on `linux-amd64` at the slowest, 34.1 seconds on `linux-arm` on physical ARM hardware. `js-wasm` does it in 8.2 seconds. Why can't `netbsd-amd64` do it in under 20 minutes?

So, no, I don't think it's just slowness.

**Comment by zoulasc** [comment]
Makes sense (that the problem lies with NetBSD or the NetBSD port). Is there a crash dump of where it is stuck?

**Comment by bradfitz** [comment]
@zoulasc, nope. Our build system just gives up on it and doesn't try to actively grab a stack. (which varies by OS, anyway) If you want to give me basic instructions for capturing some debug info we could probably wire it up to report that before failing.

**Comment by zoulasc** [comment]
I will setup a local build and try to reproduce it locally so I can give you something that works...

**Comment by bcmills** [comment]
There are two specific tests that get stuck: `cgo_test` and

**Comment by bcmills** [comment]
Still an issue. `misc/cgo/test`, `os/signal`, and `go/types` all have recent timeouts:
https://build.golang.org/log/a1652930ea41da5e9588f973a4c8a91878c00062
https://build.golang.org/log/9fcd7b26fe2a11d9883c6c18c370e36e3b3039b0
https://build.golang.org/log/a511a477b595ee1e3f444ed2b3e943bbc9e65541
https://build.golang.org/log/c0e5926e39b4704f6ff9bcfb23e41de1237ce859

There is also a `cmd/go` timeout that may or may not be related:
https://build.golang.org/log/8438c6756c49694e123c5b058e2a7903ece9d7f1

**Comment by dmitshur** [comment]
The netbsd-amd64-8_0 builder doesn't exist by now, closing.
